### PR TITLE
add a few test cases with fd_read

### PR DESCRIPTION
**Pull request by yamt**
a fix for toywasm: https://github.com/yamt/toywasm/commit/e9cc8ced70568930e526812f3706a0d7ef73953f

wamr doesn't work as of writing this.
it might be fixed when dev/interrupt_block_insn branch is done.

### CODE REVIEW

**Comment by yamt** [comment]
@abrown @loganek can you review?

File: test/testsuite/wasi_threads_exit_main_wasi_read.wat [hunk 1]
```diff
@@ -0,0 +1,58 @@
+;; When the main thread calls proc_exit, it should terminate
+;; a thread blocking in a WASI call. (poll_oneoff)
```
Comment (loganek) [inline test/testsuite/wasi_threads_exit_main_wasi_read.wat:2]:
```suggestion
;; a thread blocking in a WASI call. (fd_read)
```

Comment (yamt) [reply test/testsuite/wasi_threads_exit_main_wasi_read.wat:2]: oops, fixed. thank you.

**Comment by yamt** [comment]
@loganek @abrown any other concerns?

**Comment by loganek** [comment]
Sorry for the delay.  The change looks good to me. @abrown / anybody else: I'll go ahead and merge it as the PR was open for quite some time; feel free to open an issue if you have any comments

**Pull request merged**